### ISSUE TITLE
Users locking them selves out of the control panel when using Roll Your Own widgets

### ISSUE THREAD

**Issue by kbergha** [spicywebau/craft-odds-and-ends#24]
Hi.

There is a (high) chance of users locking themselves out of the control panel when using Roll Your Own widgets.

## How to reproduce:
If you click the "+ New widget"-button and select Roll Your Own, and then just click save (because you're curious or whatever), you then get an error saying "Couldn't save widget."

![Firefox_Screenshot_2023-03-09T09-05-22 356Z](https://user-images.githubusercontent.com/4321846/223980740-aafc924c-55bd-46ff-ac16-7e62b7c30bb8.png)

I guess the error is because the default suggestion for template (`_dashboard`) does not exist.
This is error message is understandable, and is fine in itself.

The problem is that at this stage a row is actually saved to the database with the invalid reference to `_dashboard`:

Example: 
```
mysql> select * from widgets where type like '%RollYourOwn%';
+----+--------+------------------------------------------+-----------+---------+---------------------------------------------------+---------+---------------------+---------------------+--------------------------------------+
| id | userId | type                                     | sortOrder | colspan | settings                                          | enabled | dateCreated         | dateUpdated         | uid                                  |
+----+--------+------------------------------------------+-----------+---------+---------------------------------------------------+---------+---------------------+---------------------+--------------------------------------+
|  8 |      2 | spicyweb\oddsandends\widgets\RollYourOwn |         5 |    NULL | {"title":"Roll Your Own","template":"_dashboard"} |       1 | 2023-03-09 09:05:11 | 2023-03-09 09:05:11 | b00d5bae-67f6-496e-a9a6-99885cc1b5d8 |
+----+--------+------------------------------------------+-----------+---------+---------------------------------------------------+---------+---------------------+---------------------+--------------------------------------+
1 row in set (0.00 sec)
```

Every action you then try to do in the control panel results in a "Template not found"-error - stack:

```
craft\web\twig\TemplateLoaderException: Unable to find the template “_dashboard”. in /app/vendor/craftcms/cms/src/web/twig/TemplateLoader.php:108
Stack trace:
#0 /app/vendor/craftcms/cms/src/web/twig/TemplateLoader.php(69): craft\web\twig\TemplateLoader->_resolveTemplate('_dashboard')
#1 /app/vendor/twig/twig/src/Environment.php(261): craft\web\twig\TemplateLoader->getCacheKey('_dashboard')
#2 /app/vendor/twig/twig/src/Environment.php(309): Twig\Environment->getTemplateClass('_dashboard')
#3 /app/vendor/twig/twig/src/Environment.php(277): Twig\Environment->load('_dashboard')
#4 /app/vendor/craftcms/cms/src/web/View.php(451): Twig\Environment->render('_dashboard', Array)
#5 /app/vendor/spicyweb/craft-odds-and-ends/src/widgets/RollYourOwn.php(65): craft\web\View->renderTemplate('_dashboard')
#6 /app/vendor/craftcms/cms/src/controllers/DashboardController.php(466): spicyweb\oddsandends\widgets\RollYourOwn->getBodyHtml()
#7 /app/vendor/craftcms/cms/src/controllers/DashboardController.php(104): craft\controllers\DashboardController->_getWidgetInfo(Object(spicyweb\oddsandends\widgets\RollYourOwn))
#8 [internal function]: craft\controllers\DashboardController->actionIndex()
#9 /app/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#10 /app/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#11 /app/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('index', Array)
#12 /app/vendor/craftcms/cms/src/web/Application.php(301): yii\base\Module->runAction('dashboard/index', Array)
#13 /app/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('dashboard/index', Array)
#14 /app/vendor/craftcms/cms/src/web/Application.php(286): yii\web\Application->handleRequest(Object(craft\web\Request))
#15 /app/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#16 /app/web/index.php(15): yii\base\Application->run()
#17 {main}
```

And now you're stuck without any way to disable the widget yourself.
You'll need someone with direct access to the database to delete the row.

## Runtime info:

PHP 8.2.3
Craft Pro 4.3.10
MySQL 5.7.41
Odds & Ends 4.1.1

One more piece that might be relevant - in our general config we've overridden the default extension for Twig:
`->defaultTemplateExtensions(['html.twig'])`

## Suggestions / questions:

We're actually using Odds & Ends just for the field type "Categories (multiple groups)".

Could a (partial) solution be to make which parts of this plugin to activate be configurable? 
Reading a config file in `Tools::init()` or something? With all options enabled as default to be backwards compatible.
Are you open to PRs?

**Comment by ttempleton** [comment]
Thanks for reporting that - fixed now in 4.1.2, by displaying the same notification that Craft's template UI element uses if it's set to a nonexistent template file.

Regarding making it possible to activate/deactivate parts of the plugin - that's a good point. Much appreciated if you are willing to work on a PR for that, we are open to it! 🙂 Otherwise we'll have a look at adding that soon.

**Comment by kbergha** [comment]
Thanks! The fix works perfectly.

I'll try to set aside some time to get a PR going regarding the config :)
